### ISSUE TITLE
Improve introduction documentation

### ISSUE THREAD

**Issue by rssh** [rssh/dotty-cps-async#48]
Track documentation from the point of novice and rewrite parts, which are hard to follow.  
// And not forget about experienced developers' paths, who will not want to read 'Monad-101' intro for documentation on each library.  Maybe write in addition a separate introduction library?

**Comment by ScoreUnder** [comment]
For me, I first came into this project only knowing the user side of `scala-async`, which means a "Quick Start" for users like me would hopefully amount to little more than this snippet:

```scala
import cps.*
import cps.monads.FutureAsyncMonad
```
